Mathlib/Topology/Algebra/Valued/LinearTopology.lean

```diff
@@ -0,0 +1,41 @@
+/-
+Copyright (c) 2025 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+import Mathlib.Topology.Algebra.LinearTopology
+import Mathlib.Topology.Algebra.Valued.ValuedField
+
+/-!
+# Valuation rings of valued fields have a linear topology
+
+## Main Results
+* `IsLinearTopology.of_valued`: for a valued field `K`,
+  the valuation ring `𝒪[K]` has a linear topology
+
+-/
+open Valued Filter Topology
+
+variable {K Γ₀ : Type*} [Field K] [LinearOrderedCommGroupWithZero Γ₀] [Valued K Γ₀]
+
+instance IsLinearTopology.of_valued :
+    IsLinearTopology 𝒪[K] K := by
+  rw [isLinearTopology_iff_hasBasis_open_submodule]
+  apply (hasBasis_nhds_zero K Γ₀).to_hasBasis
+  · exact fun r _ ↦ ⟨v.ltSubmodule r, isOpen_ltSubmodule _ _, subset_rfl⟩
+  · intro I hI
+    simpa [mem_nhds_zero] using hI.mem_nhds I.zero_mem
+
+instance IsLinearTopology.of_valued' :
+    IsLinearTopology 𝒪[K] 𝒪[K] := by
```

Comment:
Is there a way to make this rely less on the specific defeq of the ring of integers? And more on a predicate that asserts that `K` is the ring of fractions of some domain `𝒪` (+ suitable hyps, of course)